src/base/Context.ts

```diff
@@ -585,6 +591,60 @@ export class Context {
     return assembleResult(didRecords);
   }
 
+  /**
+   * @hidden
+   */
+  private async getIdentityClaimsFromChain(args: {
+    targets: (string | Identity)[];
+    claimTypes: ClaimType[];
+    includeExpired: boolean;
+  }): Promise<ClaimData[]> {
+    const {
+      polymeshApi: {
+        query: { identity },
+      },
+    } = this;
+
+    const { targets, claimTypes, includeExpired } = args;
+    const data: ClaimData[] = [];
+
+    const claim1stKeys = flatMap(targets, target =>
+      claimTypes.map(claimType => {
+        return {
+          target: signerToString(target),
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          claim_type: claimTypeToMeshClaimType(claimType, this),
+        };
+      })
+    );
+
+    await P.map(claim1stKeys, async claim1stKey => {
+      const entries = await identity.claims.entries(claim1stKey);
+
+      entries.forEach(
+        ([
+          key,
+          { claim_issuer: claimissuer, issuance_date: issuanceDate, expiry: rawExpiry, claim },
+        ]) => {
+          const { target } = key.args[0] as Claim1stKey;
+          const expiry = rawExpiry ? momentToDate(rawExpiry.unwrap()) : null;
+
+          if ((!includeExpired && (expiry === null || expiry > new Date())) || includeExpired) {
+            data.push({
+              target: new Identity({ did: identityIdToString(target) }, this),
+              issuer: new Identity({ did: identityIdToString(claimissuer) }, this),
+              issuedAt: momentToDate(issuanceDate),
+              expiry,
+              claim: meshClaimToClaim(claim),
+            });
+          }
+        }
+      );
+    });
```

Comment:
This approach may lead to inconsistent ordering. You should have the map function return the claim data if `(!includeExpired && (expiry === null || expiry > new Date())) || includeExpired` is true, and undefined if not, and then filter out the undefined values afterwards with lodash `compact`